packages/flutter/test/painting/image_provider_test.dart

```diff
@@ -88,30 +90,98 @@ void main() {
-    final ImageProvider imageProvider = LoadErrorImageProvider();
-    final Completer<bool> caughtError = Completer<bool>();
-    FlutterError.onError = (FlutterErrorDetails details) {
-      caughtError.complete(false);
-    };
-    final ImageStream result = imageProvider.resolve(ImageConfiguration.empty);
-    result.addListener((ImageInfo info, bool syncCall) {
-    }, onError: (dynamic error, StackTrace stackTrace) {
-      caughtError.complete(true);
+    bool uncaught = false;
+    final Zone testZone = Zone.current.fork(specification: ZoneSpecification(
+      handleUncaughtError: (Zone zone, ZoneDelegate zoneDelegate, Zone parent, Object error, StackTrace stackTrace) {
+        uncaught = true;
+      }
+    ));
+    await testZone.run(() async {
+      final ImageProvider imageProvider = LoadErrorImageProvider();
+      final Completer<bool> caughtError = Completer<bool>();
+      FlutterError.onError = (FlutterErrorDetails details) {
+        throw Error();
+      };
+      final ImageStream result = imageProvider.resolve(ImageConfiguration.empty);
+      result.addListener((ImageInfo info, bool syncCall) {
+      }, onError: (dynamic error, StackTrace stackTrace) {
+        caughtError.complete(true);
+      });
+      expect(await caughtError.future, true);
     });
-    expect(await caughtError.future, true);
+    expect(uncaught, false);
   });
 
    test('ImageProvider.resolve errors in the completer will be caught', () async {
-    final ImageProvider imageProvider = LoadErrorCompleterImageProvider();
-    final Completer<bool> caughtError = Completer<bool>();
-    FlutterError.onError = (FlutterErrorDetails details) {
-      caughtError.complete(false);
-    };
-    final ImageStream result = imageProvider.resolve(ImageConfiguration.empty);
-    result.addListener((ImageInfo info, bool syncCall) {
-    }, onError: (dynamic error, StackTrace stackTrace) {
-      caughtError.complete(true);
+    bool uncaught = false;
+    final Zone testZone = Zone.current.fork(specification: ZoneSpecification(
+      handleUncaughtError: (Zone zone, ZoneDelegate zoneDelegate, Zone parent, Object error, StackTrace stackTrace) {
+        uncaught = true;
+      }
+    ));
+    await testZone.run(() async {
+      final ImageProvider imageProvider = LoadErrorCompleterImageProvider();
+      final Completer<bool> caughtError = Completer<bool>();
+      FlutterError.onError = (FlutterErrorDetails details) {
+        throw Error();
+      };
+      final ImageStream result = imageProvider.resolve(ImageConfiguration.empty);
+      result.addListener((ImageInfo info, bool syncCall) {
+      }, onError: (dynamic error, StackTrace stackTrace) {
+        caughtError.complete(true);
+      });
+      expect(await caughtError.future, true);
     });
-    expect(await caughtError.future, true);
+    expect(uncaught, false);
+  });
+
+  test('ImageProvider.resolve errors in the completer will be caught', () async {
```

Comment:
Is this an accidental duplication of the previous test or is there something different here?

Reply:
Woops